File: Java/Vetores/comerciante.java
// Exercício: Comerciante
// Objetivo: armazenar dados de produtos, classificar os lucros percentuais e calcular os valores totais de compra, venda e lucro.

import java.util.Locale;
import java.util.Scanner;

public class comerciante {
    public static void main(String[] args) {
        
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int n, abaixo10, entre10e20, acima20;
        double lucro, totalCompra, totalVenda, lucroTotal;

        System.out.print("Serao digitados dados de quantos produtos? ");
        n = sc.nextInt();
        sc.nextLine();

        String[] nome = new String[n];
        double[] compra = new double[n];
        double[] venda = new double[n];

        for (int i = 0; i < n; i++) {
            System.out.println("Produto " + (i+1) + ":");

            System.out.print("Nome: ");
            nome[i] = sc.next();
            sc.nextLine();

            System.out.print("Preco de compra: ");
            compra[i] = sc.nextDouble();
            sc.nextLine();

            System.out.print("Preco de venda: ");
            venda[i] = sc.nextDouble();

            sc.nextLine();
        }

        abaixo10 = 0;
        entre10e20 = 0;
        acima20 = 0;

        for (int i = 0; i < n; i++) {
            lucro = (venda[i] - compra[i]) * 100 / compra[i];
            if (lucro < 10) {
                abaixo10 = abaixo10 + 1;
            }
            else if (lucro <= 20) {
                entre10e20 = entre10e20 + 1;
            }
            else {
                acima20 = acima20 + 1;
            }
        }

        totalCompra = 0;
        totalVenda = 0;

        for (int i = 0; i < n; i++) {
            totalCompra = totalCompra + compra[i];
            totalVenda = totalVenda + venda[i];
        }

        lucroTotal = 0;
        lucroTotal = totalVenda - totalCompra;

        System.out.println("RELATORIO: ");
        System.out.println();
        System.out.println("Lucro abaixo de 10%: " + abaixo10);
        System.out.println("Lucro entre 10 e 20%: " + entre10e20);
        System.out.println("Lucro acima de 20%: " + acima20);
        System.out.println("Valor total de compra: " + String.format("%.2f", totalCompra));
        System.out.println("Valor total de venda: " + String.format("%.2f", totalVenda));
        System.out.println("Lucro total: " + String.format("%.2f", lucroTotal));

        sc.close();
    }
}

File: Python/vetores/comerciante.py
# Exercício: Comerciante
# Objetivo: calcular o lucro dos produtos, classificá-los por percentual de lucro e calcular os valores totais de compra, venda e lucro.

n = int(input("Serao digitados dados de quantos produtos? "))

nome = [0 for x in range(n)]
compra = [0.0 for x in range(n)]
venda = [0.0 for x in range(n)]

for i in range(n):
    print("Produto ", i+1, ":")
    nome[i] = input("Nome: ")
    compra[i] = float(input("Preco de compra: "))
    venda[i] = float(input("Preco de venda: "))

abaixo10 = 0
entre10e20 = 0
acima20 = 0
for i in range(n):
    lucro = (venda[i] - compra[i]) * 100 / compra[i]
    if lucro < 10:
        abaixo10 = abaixo10 + 1
    elif lucro <= 20:
        entre10e20 = entre10e20 + 1
    else:
        acima20 = acima20 + 1

totalCompra = 0
totalVenda = 0
for i in range(n):
    totalCompra = totalCompra + compra[i]
    totalVenda = totalVenda + venda[i]

lucroTotal = 0
lucroTotal = totalVenda - totalCompra

print("RELATORIO: ")
print()
print("Lucro abaixo de 10%: ", abaixo10)
print("Lucro entre 10 e 20%: ", entre10e20)
print("Lucro acima de 20%: ", acima20)
print(f"Valor total de compra: {totalCompra:.2f}")
print(f"Valor total de venda: {totalVenda:.2f}")
print(f"Lucro total: {lucroTotal:.2f}")